### ISSUE TITLE
Support setting build property in gradle

### ISSUE THREAD

**Issue by murdos** [jhipster/jhipster-lite#8628]
The `SetBuildProperty` command is not implemented in `GradleCommandHandler`

- [x] https://github.com/jhipster/jhipster-lite/pull/9207
- [ ] Create agnostic `javaBuildProperties()` for gradle and maven. It will replace the `.javaDependencies().setVersion` - https://github.com/jhipster/jhipster-lite/pull/9267

**Comment by renanfranca** [comment]
Hello 😃

I want to discuss a bit before starting the implementation. Sorry for the long comment, please read 😉

### Maven
The method MavenCommandHandler.void handle(SetBuildProperty command) supports adding properties (`<properties></properties>`) and It supports adding specific properties for each profile(<`profiles><profile><properties></properties></profile></profiles>`).

I generate a jhlite project using the `spring-boot-local-profile` module with `maven` and here is the generated code:

`properties`:
```xml
  <properties>
    <java.version>21</java.version>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    <maven.version>3.9.5</maven.version>
    <junit-jupiter.version>5.10.2</junit-jupiter.version>
    <assertj.version>3.25.3</assertj.version>
    <mockito.version>5.11.0</mockito.version>
    <maven-enforcer-plugin.version>3.4.1</maven-enforcer-plugin.version>
    <compiler-plugin.version>3.12.1</compiler-plugin.version>
    <surefire-plugin.version>3.2.5</surefire-plugin.version>
    <failsafe-plugin.version>3.2.5</failsafe-plugin.version>
    <spring-boot.version>3.2.3</spring-boot.version>
    <maven-resources-plugin.version>3.3.1</maven-resources-plugin.version>
    <reflections.version>0.10.2</reflections.version>
  </properties>
```

`specific properties for profile`:
```xml
  <profiles>
    <profile>
      <id>local</id>
      <activation>
        <activeByDefault>true</activeByDefault>
      </activation>
      <properties>
        <spring.profiles.active>local</spring.profiles.active>
      </properties>
    </profile>
  </profiles>
```

As far I understand about maven, you can use both property types at resources files with the `@` character as both sufix and prefix, ex `@spring.profiles.active@` than maven will replace it with the value of the property. That is possible because the following configuration which is setup by the `maven-java` module.
`resource filtering`:
```xml
  <build>
    <resources>
      <resource>
        <directory>${basedir}/src/main/resources</directory>
        <filtering>true</filtering>
        <includes>
          <include>config/*.properties</include>
          <include>config/*.yml</include>
        </includes>
      </resource>
      <resource>
        <directory>${basedir}/src/main/resources</directory>
        <excludes>
          <exclude>config/*.properties</exclude>
          <exclude>config/*.yml</exclude>
        </excludes>
      </resource>
    </resources>
	...
```

### Gradle
@murdos : We started a discussion [here](https://github.com/jhipster/jhipster-lite/pull/9163#discussion_r1517817991) about resource filtering in gradle.

#### Profiles properties

In my mind, I think of the following solution for specific property profiles using resource filtering fixed for both `.properties` and `.yml` files.

The following resource filtering is going to be fixed at `build.gradle.kts.template`
```gradle.kts
tasks.processResources {
  filesMatching("**/*.yml", "**/*.properties") {
	filter {
	  // jhipster-needle-gradle-resource-filtering
	}
  }
  // jhipster-needle-gradle-process-resources
}
```
So, when add a specific profile property it will be added at `build.gradle.kts` and at `profile-local.gradle.kts`.


`build.gradle.kts`:
```gradle.kts
repositories {
  mavenCentral()
  // jhipster-needle-gradle-repositories
}

group = "tech.jhipster.gradleapp"
version = "0.0.1-SNAPSHOT"

val springProfilesActive by extra("")
// jhipster-needle-gradle-properties

val profiles = (project.findProperty("profiles") as String? ?: "")
  .split(",")
  .map { it.trim() }
  .filter { it.isNotEmpty() }
// jhipster-needle-profile-activation

tasks.build {
  dependsOn("processResources")
}

tasks.processResources {
  filesMatching("**/*.yml", "**/*.properties") {
	filter {
	  it.replace("@spring.profiles.active@", springProfilesActive)
	  // jhipster-needle-gradle-resource-filtering
	}
  }
  // jhipster-needle-gradle-process-resources
}
```

`src/main/resources/generator/buildtool/gradle/buildSrc/src/main/kotlin/profile-local.gradle.kts`:
```gradle.kts
plugins {
  java
  // jhipster-needle-gradle-plugins
}

dependencies {
  // Access to the `libs` object doesn't work in precompiled script plugin
  // https://github.com/gradle/gradle/issues/15383
  //testImplementation(libs.spring.boot.devtools)

  // jhipster-needle-gradle-implementation-dependencies
  // jhipster-needle-gradle-compile-dependencies
  // jhipster-needle-gradle-runtime-dependencies
  // jhipster-needle-gradle-test-dependencies
}

val springProfilesActive by extra("local")
// jhipster-needle-gradle-properties
```

#### Properties

For the case of the properties without profiles it is going to be added here:

```gradle.kts
// jhipster-needle-gradle-properties
```

But I don't know a specific user case for that.

---

Please, when you have the time, let me know what you think.

**Comment by murdos** [comment]
Hi @renanfranca,

This ticket is just about allowing to add `val springProfilesActive by extra("local")` in:

`src/main/resources/generator/buildtool/gradle/buildSrc/src/main/kotlin/profile-local.gradle.kts`
```kotlin
plugins {
  java
  // jhipster-needle-gradle-plugins
}

dependencies {
  // Access to the `libs` object doesn't work in precompiled script plugin
  // https://github.com/gradle/gradle/issues/15383
  //testImplementation(libs.spring.boot.devtools)

  // jhipster-needle-gradle-implementation-dependencies
  // jhipster-needle-gradle-compile-dependencies
  // jhipster-needle-gradle-runtime-dependencies
  // jhipster-needle-gradle-test-dependencies
}

val springProfilesActive by extra("local")
// jhipster-needle-gradle-properties
```

**The resource filtering you mentioned is not related to this ticket.**

However my intention was to go a step further, and allow declaring java build properties also outside profiles, with the following API (for maven and gradle):
```java
  return moduleBuilder(properties)
      .javaBuildProperties()
        .add("node.version", "v" + npmVersions.getNodeVersion())
        .add("npm.version", npmVersions.get("npm", NpmVersionSource.COMMON).get())
        .and()
```

This should allow removing `.setVersion(..)`API  from  `.javaDependencies()` (it works only for maven where a version is a properties, while for gradle a version is not defined with a property, but in toml catalog version):
```java
  return moduleBuilder(properties)
      .javaDependencies()
        .setVersion(javaDependencyVersion("node", "v" + npmVersions.getNodeVersion()))
        .setVersion(javaDependencyVersion("npm", npmVersions.get("npm", NpmVersionSource.COMMON).get()))
        .and()
```

**Comment by renanfranca** [comment]
Thank you for the fast response 😃

> This ticket is just about allowing to add val springProfilesActive by extra("local") in:
src/main/resources/generator/buildtool/gradle/buildSrc/src/main/kotlin/profile-local.gradle.kts

I am going to implement it in a separate pull request.

> The resource filtering you mentioned is not related to this ticket.

I will follow your lead, please, give me a heads-up when the time for implementing the resource filtering comes 😉

> However my intention was to go a step further, and allow declaring java build properties also outside profiles, with the following API (for maven and gradle):

> This should allow removing .setVersion(..)API from .javaDependencies() (it works only for maven where a version is a properties, while for gradle a version is not defined with a property, but in toml catalog version):

I will do it in a different pull request. 🖖

**Comment by renanfranca** [comment]
>  Create agnostic javaBuildProperties() for gradle and maven. It will replace the .javaDependencies().setVersion

I am going to work on this issue on monday 👍
